### DISCUSSION TITLE
Inline variable refactor feature

### DISCUSSION THREAD

**Discussion by jaroslawk** [microsoft/pylance-release#2594] [Enhancement]
I am a daily user of PyCharm and was looking for an alternative tool to switch over.
Absolute basic refactoring which I use very often is:

* extract variable
* inline variable

The first one exists but the second is missing.
What are the chances to introduce it into VS Code?

This need was already reported: https://github.com/microsoft/vscode-python/issues/144 
but was not picked up

**Comment by judej** [comment]
Moving this issue to discussion as an enhancement request for comments and upvotes.

**Comment by aleksanb** [comment]
Inline variable refactor feature

The vscode-python plugin allows for extracting expressions into variables and functions, but not the reverse, inlining them into its callsites.

An earlier issue about this ages ago was closed https://github.com/microsoft/vscode-python/issues/144, but the need for inlining variables still exists, and I'd love for this feature to be added to vscode-python.


My issue https://github.com/microsoft/pylance-release/issues/7022 was closed so i'm posting here instead.

**Reply by aleksanb** [reply]
Seems like there's been no progress on this for 4 years?

**Reply by rchiodo** [reply]
We generally prioritize things based on up votes. I think this has only 3? so that's the likely reason why it's not implemented

**Reply by T3sT3ro** [reply]
Consider that a vast majority of people may be too lazy to look for that issue, report and upvote it. It's a terrible pain to inline vars by hand.

**Reply by aleksanb** [reply]
It's a quite common and useful feature.
As the reverse exists (extract) it would be nice to have the matching inline one as well.

Typescript in vscode has support for this amongst others, and i use it quite often.
<img width="433" alt="image" src="https://github.com/user-attachments/assets/1ab7f0b5-537d-4477-b39e-962420ab1e21" />

**Reply by T3sT3ro** [reply]
And any JetBrains tool I can think of as well :+1:
